### PR TITLE
Always return when err is not nil.

### PR DESCRIPTION
**Pull request by brendandburns**

### CODE REVIEW

**Comment by spf13** [comment]
LGTM

**Comment by spf13** [comment]
Merged as 11b7cf8

File: flag.go [hunk 1]
```diff
@@ -498,6 +498,7 @@ func (f *FlagSet) parseShortArg(s string, args []string) (a []string, err error)
 			if len(args) == 0 {
```
Comment (jlowdermilk) [inline flag.go]: This block is now redundant.

File: flag.go [hunk 2]
```diff
@@ -551,6 +552,9 @@ func (f *FlagSet) parseArgs(args []string) (err error) {
 		} else {
 			args, err = f.parseShortArg(s, args)
 		}
+		if err != nil {
```
Comment (jlowdermilk) [inline flag.go]: Move into above block?

**Pull request closed without merging**